### DISCUSSION TITLE
How to get selected upstream assets

### DISCUSSION THREAD

**Discussion by yan91083** [dagster-io/dagster#18316] [Q&A] [answered]
I have 3 upstream assets(files) and 1 downstream asset(training model). How can I select the asset/assets on the UI and pass it to the downstream asset to train language model?

**Comment by smackesey** [comment]
If you have assembled the asset dependencies correctly you should be able to just hit "Materialize All". But your question isn't entirely clear to me-- can you post some code if the above answer is insufficient?

**Reply by yan91083** [reply]
Thanks for replying. 
example code:
@asset 
def file1() -> str:
    return "file1"
@asset
def file2() -> str:
    return "file2"
@asset
def file3() -> str:
    return "file3"

@asset
def train(context: AssetExecutionContext, file1: str, file2: str, file3: str):
    #get the selected file/files
    
If I chose file1 & train assets and hit "Materialize all", how can I know which file is selected in the train asset?

**Reply by smackesey** [reply, marked as answer]
Is the idea that a given run of `train` is only supposed to use one of `file1`, `file2`, `file3`, not all of them at once? Your current setup uses them all.

If you want to select one at a time you should consolidate the three upstream assets into a single partitioned asset `file`, with the three files under different partition keys. Then you can select one by executing against a specific partition key.

Read more about partitions here: https://docs.dagster.io/concepts/partitions-schedules-sensors/partitions#partitions
